File: abc265/1.cpp
#include <iostream>
using namespace std;

int main(){

    int x, y, n, ans;

    cin >> x >> y >> n;

    if ( y / 3 > x ) {
        ans = x * n;
    } else {
        ans = y * ( n / 3 );
        ans += x * ( n % 3);
    }

    cout << ans << endl;

    return(0);
}

File: abc265/1.swift
let input = readLine()!.split(separator: " ")

let x = Int(input[0])!
let y = Int(input[1])!
let n = Int(input[2])!

let ans: Int

if ( y / 3 < x ) {
    ans = ( n / 3 ) * y + ( n % 3 ) * x
} else {
    ans = n * x
}

print(ans)